### PR TITLE
[test] Add a test for simple_derivative example

### PR DESCRIPTION
**Pull request by Tiny-Box**
Related issue = #3580

like ut about autodiff/regression, this is also a UT that outputs images directly

### CODE REVIEW

File: tests/python/examples/autodiff/test_simple_derivative.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    except OSError as exc: # Python >2.5
+        if exc.errno == EEXIST and path.isdir(dir_path):
+            pass
+        else: raise
+
+
+def pic_simple_derivative(result_dir):
+    from taichi.examples.autodiff.simple_derivative import (initialize,
+                                                            xs, ys, grad_xs)
+    from matplotlib import pyplot as plt
```
Comment (ailzhang) [inline tests/python/examples/autodiff/test_simple_derivative.py:20]: It's likely that we don't have matplotlib in CI test environment yet. So please feel free to add it in https://github.com/taichi-dev/taichi/blob/master/requirements_test.txt if you need it!

File: tests/python/examples/autodiff/test_simple_derivative.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import argparse
+
+
+def mkdir_p(dir_path):
```
Comment (ailzhang) [inline tests/python/examples/autodiff/test_simple_derivative.py:4]: Yea this one should be moved to shared https://github.com/taichi-dev/taichi/blob/master/tests/test_utils.py?

**Review by ailzhang — COMMENTED** [review]
Thanks for working on this!

File: tests/python/examples/autodiff/test_simple_derivative.py (hunk 1 above)
Comment (strongoier) [reply tests/python/examples/autodiff/test_simple_derivative.py:20]: The problem here is that the original example uses matplotlib, so importing from the example will result in error. I suggest adding it to `requirements_tests.txt`.

File: tests/python/examples/autodiff/test_simple_derivative.py (hunk 2 above)
Comment (strongoier) [reply tests/python/examples/autodiff/test_simple_derivative.py:4]: Yea you can fix #4322 first :-)

File: tests/python/examples/autodiff/test_simple_derivative.py (hunk 1 above)
Comment (Tiny-Box) [reply tests/python/examples/autodiff/test_simple_derivative.py:20]: get it, I will add matplotlib to `requirements_test.txt` in #4322 first

**Comment by strongoier** [comment]
@Tiny-Box You can now fix this PR :-)

**Review by strongoier — COMMENTED** [review]
Did you forget to add `test_simple_derivative`?

**Comment by Tiny-Box** [comment]
Oh, I forget it! I will add it today.

**Review by strongoier — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**